Handle transaction status updates in GUI caused by islocks and chainlocks on a per-tx basis

UdjinM6 opened the pull request:
This fixes only the GUI part. Another part of the slowness for huge wallets is the wallet itself (which reveals itself whenever you try to create a tx) but this patch removes most major freezes already and GUI is much more responsive now.

codablock commented:
Hmm, it looks like the way the need for status updates is handled with IS locks was flawed from day one (f8f14351475ac70f5fd7f47d96683117f841b6eb). With this type of check we would update all TXs even if just a single one has changed the IS lock status. Here is an alternative implementation:
ee76c652db67d5cc89d2c0e79d296670588dd8b9
c4b46c7b8c6752853024d5798cbde7e59aa8d18f

UdjinM6 commented:
Applied your patches, extracted `Fix updateConfirmations slowness for huge wallets` commit into its own PR #3154 and renamed this one to better match actual changes.

UdjinM6 commented:
Closing in fav of #3155
